src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -34,6 +36,34 @@ class WorkflowDataConditionGroupType(StrEnum):
     WORKFLOW_TRIGGER = "workflow_trigger"
 
 
+def create_workflow_fire_histories(workflows: set[Workflow], event_data: WorkflowEventData) -> None:
+    # Create WorkflowFireHistory objects for triggered workflows
+    fire_histories = [
+        WorkflowFireHistory(
+            workflow=workflow, group=event_data.event.group, event_id=event_data.event.event_id
+        )
+        for workflow in workflows
+    ]
+    WorkflowFireHistory.objects.bulk_create(fire_histories)
+
+
+def update_workflow_fire_histories(
+    actions_to_fire: BaseQuerySet[Action], event_data: WorkflowEventData
+) -> None:
```

Comment:
I feel like this method should live in `processors/action` because we are taking a list of them actions and updating a state to say that we have triggered actions. in the future, this gives us a specific hook to update for the notification platform tracking.